tests/unit/zhmcclient/test_resource.py

```diff
@@ -642,3 +643,778 @@ def test_get_properties_local(
 
             assert value == exp_value, \
                 "Unexpected property value for '{}'".format(name)
+
+
+class TestPropertyMethodsMocked(object):
+    """
+    All tests of property methods that need mocked resources.
+    """
+
+    RESOURCE_OID = 'res-oid'
+    RESOURCE_NAME = 'res-name'
+    RESOURCE_DESC = 'Resource Description'
+
+    def setup_method(self):
+        """
+        Setup that is called by pytest before each test method.
+
+        Set up a faked session, and add a faked CPC in DPM mode without any
+        child resources.
+        """
+        # pylint: disable=attribute-defined-outside-init
+
+        self.session = FakedSession('fake-host', 'fake-hmc', '2.13.1', '1.8')
+        self.client = Client(self.session)
+
+        self.faked_cpc = self.session.hmc.cpcs.add({
+            'object-id': 'fake-cpc1-oid',
+            # object-uri is set up automatically
+            'parent': None,
+            'class': 'cpc',
+            'name': 'fake-cpc1-name',
+            'description': 'CPC #1 (DPM mode)',
+            'status': 'active',
+            'dpm-enabled': True,
+            'is-ensemble-member': False,
+            'iml-mode': 'dpm',
+            'machine-type': '2964',  # z13
+            'machine-model': 'N10',
+        })
+        self.cpc = self.client.cpcs.find(name='fake-cpc1-name')
+
+    def add_standard_hipersocket(self):
```

Comment:
Out of curiosity, why are you setting up a hipersocket adapter?

Reply:
Because that is an example of a creatable resource type that does not support the "properties" query parameter. See the docstring of the method.